src/serializers/fields.rs

```diff
@@ -136,50 +145,21 @@ impl GeneralFieldsSerializer {
             }
         }
     }
-}
-
-macro_rules! option_length {
-    ($op_has_len:expr) => {
-        match $op_has_len {
-            Some(ref has_len) => has_len.len(),
-            None => 0,
-        }
-    };
-}
 
-impl_py_gc_traverse!(GeneralFieldsSerializer {
-    fields,
-    computed_fields
-});
-
-impl TypeSerializer for GeneralFieldsSerializer {
-    fn to_python(
+    pub fn main_to_python<'py>(
         &self,
-        value: &PyAny,
-        include: Option<&PyAny>,
-        exclude: Option<&PyAny>,
-        extra: &Extra,
-    ) -> PyResult<PyObject> {
-        let py = value.py();
-        // If there is already a model registered (from a dataclass, BaseModel)
-        // then do not touch it
-        // If there is no model, we (a TypedDict) are the model
-        let td_extra = Extra {
-            model: extra.model.map_or_else(|| Some(value), Some),
-            ..*extra
-        };
-        let (main_dict, extra_dict) = if let Some(main_extra_dict) = self.extract_dicts(value) {
-            main_extra_dict
-        } else {
-            td_extra.warnings.on_fallback_py(self.get_name(), value, &td_extra)?;
-            return infer_to_python(value, include, exclude, &td_extra);
-        };
-
+        py: Python<'py>,
+        main_iter: impl Iterator<Item = PyResult<(&'py PyAny, &'py PyAny)>>,
```

Comment:
generic function for building a dict from an iterator, so we can use both in the `GeneralFieldsSerializer` serializer and directly from the dataclass serializer.